nucypher/blockchain/eth/sol/compile.py

```diff
@@ -74,20 +92,43 @@ def install_compiler(self, version: str=None):
         Installs the specified solidity compiler version.
         https://github.com/ethereum/py-solc#installing-the-solc-binary
         """
-        version = version if version is not None else self.__default_version
+        version = version if version is not None else self.__default_compiler_version
         return install_solc(version, platform=None)  # TODO: fix path
 
     def compile(self) -> dict:
+        interfaces = dict()
+        for root_source_dir, other_source_dirs in self.source_dirs:
+            if root_source_dir is None:
+                self.log.warn("One of the root directories is None")
+                continue
+
+            raw_interfaces = self._compile(root_source_dir, other_source_dirs)
+            for name, data in raw_interfaces.items():
+                # Extract contract version from docs
+                version_search = re.search(r'\"details\":\".*?\|(v\d+\.\d+\.\d+)\|.*?\"', data['devdoc'])
```

Comment:
Can you expand this regex into the verbose format?  

Example:

```
p = re.compile(r"""

(\+\d{1,2})?               # Optional Country Code +1 or perhaps +22
(-|\()?                    # Optional hyphen or open parenths for the area code.
(\d{10}                    # Try and match ten digits exactly
|                          # Else
(\d{3}                     # Match three digits
[\-|\)]{0,2})              # Hypen or close parenths zero through 2 times (to account for -))  
{2}                        # Repeat to match the next three digits
\d{4})                     # The remaining four digits
$                          # Ensure nothing else follows 


""", re.VERBOSE)

```

Reply:
✔️